embedded-keycloak/src/test/java/com/playtika/test/keycloak/EmbeddedKeycloakBootstrapConfigurationWithRealmTest.java

```diff
@@ -0,0 +1,100 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.playtika.test.keycloak;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+@Slf4j
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+    classes = KeycloakTestApplication.class
+)
+@ActiveProfiles("realm")
+public class EmbeddedKeycloakBootstrapConfigurationWithRealmTest {
+
+    @Autowired
+    private ConfigurableEnvironment environment;
+
+    @Autowired
+    private KeycloakContainer keycloakContainer;
+
+    @Test
+    public void keycloakContainerAvailable() {
+        assertThat(keycloakContainer).isNotNull();
+    }
+
+    @Test
+    public void shouldGetTestRealmInfoFromKeycloak() {
+        RestTemplate restTemplate = new RestTemplate();
+        String baseUrl = environment.getProperty("embedded.keycloak.auth-server-url");
+
+        RealmInfo realmInfo = restTemplate.getForObject(
+            baseUrl + "/realms/test-realm",
+            RealmInfo.class
+        );
+
+        assertThat(realmInfo.getRealm()).isEqualTo("test-realm");
+    }
+
+    @Test
+    public void shouldGetAccessTokenFromKeycloak() {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+
+        String clientId = environment.getProperty("keycloak.resource");
+        String clientSecret = environment.getProperty("keycloak.credentials.secret");
+
+        MultiValueMap<String, String> map = new LinkedMultiValueMap<>();
+        map.add("client_id", clientId);
+        map.add("client_secret", clientSecret);
+        map.add("grant_type", "password");
+        map.add("username", "keycloak-test");
+        map.add("password", "123Start!");
+
+        String baseUrl = environment.getProperty("embedded.keycloak.auth-server-url");
+
+        KeyCloakToken token = restTemplate
```

Comment:
correct me if I'm wrong, but all tests tests  keycloak container via vanilla REST, 
can we test it also with spring extention lib somehow?

Reply:
Yeah, but will need some time to figure out how to properly do that.